src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -332,13 +393,25 @@ internal override async ValueTask<QuicStreamProvider> AcceptStreamAsync(Cancella
         internal override QuicStreamProvider OpenUnidirectionalStream()
         {
             ThrowIfDisposed();
-            return new MsQuicStream(_state, QUIC_STREAM_OPEN_FLAGS.UNIDIRECTIONAL);
+            lock (_state)
+            {
+                var stream = new MsQuicStream(_state, QUIC_STREAM_OPEN_FLAGS.UNIDIRECTIONAL);
+                // once stream is created, it will call RemoveStream on disposal.
+                _state.StreamCount++;
```

Comment:
Since stream is the one removing itself on disposal, would it be logical for stream to also add itself in ctor?

Reply:
I was thinking about it but I did not do it. I like the symmetry e.g. ctor/disposal. I'll make the update during next round unless somebody object.

Reply:
updated.